crates/bevy_render/src/pipeline/pipeline_layout.rs

```diff
@@ -25,10 +26,15 @@ impl PipelineLayout {
                         for shader_binding in shader_bind_group.bindings.iter() {
                             if let Some(binding) = bind_group
                                 .bindings
-                                .iter()
+                                .iter_mut()
                                 .find(|binding| binding.index == shader_binding.index)
                             {
-                                if binding != shader_binding {
+                                binding.shader_stage |= shader_binding.shader_stage;
+                                // Not sure we need to panic anymore here..
```

Comment:
If we don't panic when binding indices dont match across shader stages, then im pretty sure wgpu will panic instead. eventually this should return an Error instead, which would make this recoverable.